File: Composition/CS/Composition/Algorithm/Inheritance/PointsAggregator.cs
using System.Collections.Generic;

namespace Algorithm.Inheritance
{
    public abstract class PointsAggregator
    {
        protected PointsAggregator(IEnumerable<Measurement> measurements)
        {
            Measurements = measurements;
        }

        public virtual Measurement Aggregate()
        {
            var measurements = Measurements;
            measurements = FilterMeasurements(measurements);
            return AggregateMeasurements(measurements);
        }

        protected abstract IEnumerable<Measurement> FilterMeasurements(IEnumerable<Measurement> measurements);
        protected abstract Measurement AggregateMeasurements(IEnumerable<Measurement> measurements);
        
        protected readonly IEnumerable<Measurement> Measurements;
    }        
}

File: Composition/Java/Composition/src/main/java/algorithm/inhereitance/PointsAggregator.java
package algorithm.inhereitance;

import java.util.Collection;

import algorithm.Measurement;

public abstract class PointsAggregator {

	protected abstract Collection<Measurement> filterMeasurements();
	protected abstract Measurement aggregateMeasurements();

	protected Collection<Measurement> measurements;

	protected PointsAggregator(Collection<Measurement> measurements) {
		this.measurements = measurements;
	}

	public Measurement aggregate() {
		measurements = filterMeasurements();
		return aggregateMeasurements();
	}

}
